### ISSUE TITLE
capability to write workflow structure to a json file 

### ISSUE THREAD

**Issue by vivek-bala** [radical-cybertools/radical.entk#190]
once the workflow is created using PST, it will be useful to have function that dumps all the values to a JSON file on the client

**Comment by vivek-bala** [comment]
This will be part of the next point release. Removing 0.6 tag.

**Comment by jdakka** [comment]
Update: 
Started working on this. Created new branch `feature/json`. I've created `pprints` in the app manager to capture the `PST` model. Example:

```
'Total of 1 pipeline(s)'
'Pipeline 0,  contains 1 stage(s)'
1
'Stage 0,  contains 1 task(s)'
'Tasks: my-first-task'
```
These outputs will become entries in the `.json` database, along with the `res_dict`. I was thinking we can also add in the `pipeline/stage/task.uid` along with the `pipeline/stage/task.name`

@vivek-bala, thoughts on what else would be useful for use-cases?

**Comment by andre-merzky** [comment]
Quick note that we can capture workloads on RP level, by setting `RADICAL_PILOT_RECORD_SESSION`.  We probably cannot derive the EnTK entities from that - but you may want to consider to use a similar mechanism so that we capture the layers simultaneously.

This is implemented like this (here for recording units, similar code exists for session and pilot): https://github.com/radical-cybertools/radical.pilot/blob/devel/src/radical/pilot/unit_manager.py#L746

**Comment by vivek-bala** [comment]
Can you push the branch and put a PR? I'll be able to see the diff and comment on what else might be required.

**Comment by vivek-bala** [comment]
FWIW all attributes of the Task object need to be recorded. For the Pipeline and Stage object, we need (as a minimum) the uids and the uids of its children to be able to find the relationship between the tasks.

**Comment by jdakka** [comment]
Thanks I'll look at the implementation from @andre-merzky and start building up from there.

**Comment by vivek-bala** [comment]
How's the implementation going @jdakka? This ideally needs to be part of the next release on Friday.

**Comment by vivek-bala** [comment]
bump

**Comment by jdakka** [comment]
[pull request](https://github.com/radical-cybertools/radical.entk/pull/224)

**Comment by vivek-bala** [comment]
This is implemented in 0.6.2
